pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginClassLoader.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.plugin;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Enumeration;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+
+public class PluginClassLoader extends URLClassLoader {
+
+  private final ClassLoader sysClzLoader;
+
+  public PluginClassLoader(URL[] urls, ClassLoader parent) {
+    super(urls, parent);
+    sysClzLoader = getSystemClassLoader();
+    URLClassLoader classLoader = (URLClassLoader) ClassLoader.getSystemClassLoader();
+    Method method = null;
+    try {
+      method = URLClassLoader.class.getDeclaredMethod("addURL", URL.class);
+
+    } catch (NoSuchMethodException e) {
+      //this should never happen
+      ExceptionUtils.rethrow(e);
+    }
+    method.setAccessible(true);
+    for (URL url : urls) {
+      try {
+        method.invoke(classLoader, url);
+      } catch (Exception e) {
+        ExceptionUtils.rethrow(e);
+      }
+    }
+  }
+
+  @Override
+  protected Class<?> loadClass(String name, boolean resolve)
+      throws ClassNotFoundException {
+    if (name.endsWith("LocalFileSystem")) {
+      System.out.println("name = " + name);
+    }
+    System.out.println("PluginClassLoader.loadClass: " + name);
```

Comment:
Are these intentional logs, or left over from debugging? Also, does it make sense to use Logger?